### PR TITLE
zs-apc-spdu-ctl: init at 0.0.2

### PR DESCRIPTION
**Pull request by fogti**
###### Motivation for this change

Adding my own APC SPDU management utility (uses SNMP).
Also adds my own library against which this tool links.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114283` at bf41843a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/9c7ab09fce321765b8467ec86d393d44">libowlevelzs</a></li></li>
    <li><a href="https://gist.github.com/5b74d29bb65ce33809b5440f27f2367d">zs-apc-spdu-ctl</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/libraries/libowlevelzs/default.nix:18:3:
```
   |
18 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/tools/networking/zs-apc-spdu-ctl/default.nix:25:3:
```
   |
25 |   patchPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/libowlevelzs/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  preBuild = ''
+    cmakeFlagsArray+=(
+      "-DBUILD_SHARED_LIBS=true"
+      "-DINSTALL_LIB_DIR=$out/lib"
+      "-DINSTALL_INCLUDE_DIR=$out/include/zs/ll"
+    )
+  '';
+
+  enableParallelBuilding = true;
```
Comment (r-rmcgibbo) [inline pkgs/development/libraries/libowlevelzs/default.nix:32]: Not required: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unnecessary-parallel-building.md

File: pkgs/development/libraries/libowlevelzs/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ cmake,
+  fetchurl,
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libowlevelzs/default.nix:2]: Please move the , to the beginning of the line.

File: pkgs/development/libraries/libowlevelzs/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  src = fetchurl {
+    url = "https://ytrizja.de/distfiles/${pname}-${version}.tar.gz";
+    sha256 = "49892b9ccd0943c23dc5af842f929ce9467ea50abd6a0e985f506464e8e8bfe0";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  postPatch = ''
+    patchShebangs config_relink.sh
+    sed -i -e "s/cstdint/stdint.h/g" src/hash.hpp
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libowlevelzs/default.nix:20]: Please use substituteInPlace.

File: pkgs/tools/networking/zs-apc-spdu-ctl/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ cmake,
+  fetchFromGitHub,
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/zs-apc-spdu-ctl/default.nix:2]: Please love the , to the beginning of the line.

File: pkgs/tools/networking/zs-apc-spdu-ctl/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    repo = "zs-apc-spdu-ctl";
+    rev = "v${version}";
+    sha256 = "Xbv/DyoKEjdJ/a5jk/K+MZpMgdZTQmVwHAwDIWcbdMo=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ fping libowlevelzs net-snmp ];
+
+  postPatch = ''
+    sed -i -e "s#/usr/sbin/fping#${lib.makeBinPath [fping]}/fping#g" src/confent.cxx
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/zs-apc-spdu-ctl/default.nix:26]: Please use substituteInPlace.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114283` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>libowlevelzs</li>
<li>zs-apc-spdu-ctl</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libowlevelzs:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'cstdint' doesn't match anything in file 'src/hash.hpp'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>
</details>

File: pkgs/development/libraries/libowlevelzs/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    url = "https://ytrizja.de/distfiles/${pname}-${version}.tar.gz";
+    sha256 = "49892b9ccd0943c23dc5af842f929ce9467ea50abd6a0e985f506464e8e8bfe0";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  postPatch = ''
+    patchShebangs config_relink.sh
+    substituteInPlace src/hash.hpp \
+      --replace cstdint stdint.h \
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libowlevelzs/default.nix:21]: substituteStream(): WARNING: pattern 'cstdint' doesn't match anything in file 'src/hash.hpp'

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114283` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>libowlevelzs: <a href="https://termbin.com/91nt">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/fe0c1ed7a039fc6dbe392b8cc20c0d11/raw/index.html">pretty log </a></li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libowlevelzs:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern 'cstdint' doesn't match anything in file 'src/hash.hpp'
```

Please check the offending substituteInPlace for typos or changes in source.
notice: no-build-output
‘libowlevelzs’ has not yet been built. Checks that rely on the presence of build logs are skipped.
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114283` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libowlevelzs</li>
</ul>
</details>

**Comment by zseri** [comment]
ok... I'm a bit suprised that the review *only* built `libowlevelzs`, when there are in fact two packages introduced by this PR.

**Comment by SuperSandro2000** [comment]
> ok... I'm a bit suprised that the review _only_ built `libowlevelzs`, when there are in fact two packages introduced by this PR.

The other package is not available on darwin.

**Comment by zseri** [comment]
ok.

**Pull request merged**